src/reconcile/rec_write.c

```diff
@@ -410,10 +410,8 @@ __rec_write_page_status(WT_SESSION_IMPL *session, WT_RECONCILE *r)
      * Track the tree's maximum transaction ID (used to decide if it's safe to discard the tree) and
      * maximum timestamp.
      */
-    if (WT_TXNID_LT(btree->rec_max_txn, r->max_txn))
-        btree->rec_max_txn = r->max_txn;
-    if (btree->rec_max_timestamp < r->max_ts)
-        btree->rec_max_timestamp = r->max_ts;
```

Comment:
Is this a correctness bug because it implies that `btree->rec_max_foo` can go backwards? If so, can you make sure this ends up in the queue for 8.0 (and backports). Also, we don't have release semantics here, which at least seems questionable...

Reply:
This is a correctness bug, so I will request a backport to 8.0 assuming the change is zero risk.